src/OVAL/oval_variable.c

```diff
@@ -84,6 +84,85 @@ typedef struct {
 	} unused;
 } oval_variable_UNKNOWN_t;
 
+typedef struct oval_variable_possible_value {
+	char *hint;
+	char *value;
+} oval_variable_possible_value_t;
+
+typedef struct oval_variable_possible_restriction {
+	oval_operator_t operator;
+	char *hint;
+	struct oval_collection *restrictions;
+} oval_variable_possible_restriction_t;
+
+typedef struct oval_variable_restriction {
+	oval_operation_t operation;
+	char *value;
+} oval_variable_restriction_t;
+
+struct oval_variable_possible_value *oval_variable_possible_value_new(const char *hint, const char *value)
+{
+	struct oval_variable_possible_value *pv;
+	pv = oscap_alloc(sizeof(oval_variable_possible_value_t));
+	pv->hint = oscap_strdup(hint);
+	pv->value = oscap_strdup(value);
+	return pv;
+}
+
+void oval_variable_possible_value_free(struct oval_variable_possible_value *pv)
+{
+	if (pv != NULL) {
+		oscap_free(pv->hint);
+		oscap_free(pv->value);
+		oscap_free(pv);
+	}
+}
+
+struct oval_variable_possible_restriction *oval_variable_possible_restriction_new(oval_operator_t operator, const char *hint)
+{
+	struct oval_variable_possible_restriction *pr;
+	pr = oscap_alloc(sizeof(oval_variable_possible_restriction_t));
+	pr->operator = operator;
+	pr->hint = oscap_strdup(hint);
+	pr->restrictions = oval_collection_new();
+	return pr;
+}
+
+void oval_variable_possible_restriction_free(struct oval_variable_possible_restriction *pr)
+{
+	if (pr != NULL) {
+		oscap_free(pr->hint);
+		if (pr->restrictions != NULL) {
```

Comment:
I think it can't, because I allocate it in the constructor. I think I can remove the if condition.